[Bug] Pin dependencies

There is currently a [bug](https://github.com/streamich/unionfs/issues/765) in the latest release of https://github.com/streamich/unionfs. The `@temporalio/worker` package does not pin this dependency (currently set to `"unionfs": "^4.4.0"`) so the package is unusable causing a company-wide outage for us. This same thing has happened on multiple occasions over the last few months.

This sdk is a core piece of infrastructure and should not suddenly break when indirect dependencies are updated. 

Error code from offending package:
```
Error: Cannot find module 'node_modules/unionfs/lib/index.js'. Please verify that the package.json has a valid "main" entry
```

--- a/packages/worker/package.json
+++ b/packages/worker/package.json
@@ -30,7 +30,7 @@
     "source-map": "^0.7.4",
     "source-map-loader": "^4.0.0",
     "swc-loader": "^0.2.3",
-    "unionfs": "^4.4.0",
+    "unionfs": "^4.5.1",
     "webpack": "^5.75.0"
   },
   "bugs": {

